std/string.d

```diff
@@ -5798,36 +5798,57 @@ unittest
  * function, or any of the conversion functions.
  *
  * Params:
- *     s = the string to check
+ *     s = the string or random access range to check
  *     bAllowSep = accept separator characters or not
  *
  * Returns:
  *     $(D bool)
  */
-bool isNumeric(S)(S s, bool bAllowSep = false) if (isSomeString!S)
+bool isNumeric(S)(S s, bool bAllowSep = false) if (isSomeString!S ||
+    (isRandomAccessRange!S &&
+    hasSlicing!S &&
```

Comment:
Is there anything that makes it unfeasible to accept mere input ranges? I understand if you don't want to do that work in the first step, and we can relax the constraints later on, but it should be possible, right?